sympy/polys/euclidtools.py

```diff
@@ -1880,6 +1901,11 @@ def dmp_cancel(f, g, u, K, include=True):
 
         cq, f = dmp_clear_denoms(f, u, K0, K, convert=True)
         cp, g = dmp_clear_denoms(g, u, K0, K, convert=True)
+        # domain is now K
+    elif K.is_ZZ:
+        # remove gcd
+        cp, f = dmp_ZZ_gcd(f, K)
+        cq, g = dmp_ZZ_gcd(g, K)
```

Comment:
This should not special case ZZ rather than something more general.

Also we have `dmp_ground_primitive` which does this.